packages/atomic/src/components/commerce/facets/atomic-commerce-numeric-facet/atomic-commerce-numeric-facet.tsx

```diff
@@ -0,0 +1,277 @@
+} from '@coveo/headless/commerce';
+import {Component, Element, h, Listen, Prop, State} from '@stencil/core';
+import {FocusTargetController} from '../../../../utils/accessibility-utils';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {shouldDisplayInputForFacetRange} from '../../../common/facets/facet-common';
+import {FacetInfo} from '../../../common/facets/facet-common-store';
+import {FacetContainer} from '../../../common/facets/facet-container/facet-container';
+import {FacetGuard} from '../../../common/facets/facet-guard';
+import {FacetHeader} from '../../../common/facets/facet-header/facet-header';
+import {FacetPlaceholder} from '../../../common/facets/facet-placeholder/facet-placeholder';
+import {formatHumanReadable} from '../../../common/facets/numeric-facet/formatter';
+import {NumericFacetValueLink} from '../../../common/facets/numeric-facet/value-link';
+import {NumericFacetValuesContainer} from '../../../common/facets/numeric-facet/values-container';
+import {
+  defaultNumberFormatter,
+  NumberFormatter,
+} from '../../../common/formats/format-common';
+import {initializePopover} from '../../../search/facets/atomic-popover/popover-type';
+import {CommerceBindings as Bindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import type {Range} from '../facet-number-input/atomic-commerce-facet-number-input';
+
+/**
+ * The `atomic-commerce-numeric-facet` component is responsible for rendering a commerce facet that allows the user to filter products using numeric ranges.
+ *
+ * @internal
+ */
+@Component({
+  tag: 'atomic-commerce-numeric-facet',
+  styleUrl: './atomic-commerce-numeric-facet.pcss',
+  shadow: true,
+})
+export class AtomicCommerceNumericFacet
+  implements InitializableComponent<Bindings>
+{
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() private host!: HTMLElement;
+
+  @State() private range?: Range;
+
+  @BindStateToController('facet')
+  @State()
+  public facetState!: NumericFacetState;
+
+  @State() public error!: Error;
+
+  private summary!: ListingSummary | SearchSummary;
+  private isCollapsed = false;
+  private manualRanges: (NumericRangeRequest & {label?: string})[] = [];
+  private formatter: NumberFormatter = defaultNumberFormatter;
+
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop({reflect: true}) public facet!: NumericFacet;
+
+  private headerFocus?: FocusTargetController;
+
+  private get focusTarget(): FocusTargetController {
+    if (!this.headerFocus) {
+      this.headerFocus = new FocusTargetController(this);
+    }
+    return this.headerFocus;
+  }
+
+  public initialize() {
+    this.initializeSummary();
+    this.registerFacetToStore();
+  }
+
+  private initializeSummary() {
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      this.summary = buildListingSummary(this.bindings.engine);
```

Comment:
should the summary controller be passed down as a prop from the parent search interface rather than initializing it here?